core/src/test/java/org/apache/iceberg/TestIncrementalDataTableScan.java

```diff
@@ -51,8 +64,8 @@ public void testAppends() {
     add(table.newAppend(), files("C"));
     add(table.newAppend(), files("D"));
     add(table.newAppend(), files("E")); // 5
-    Assert.assertEquals(Sets.newHashSet("B", "C", "D", "E"), appendsBetweenScan(1, 5));
-    Assert.assertEquals(Sets.newHashSet("C", "D", "E"), appendsBetweenScan(2, 5));
+    filesMatch(Lists.newArrayList("B", "C", "D", "E"), appendsBetweenScan(1, 5));
```

Comment:
Moved from Set to List, as Set could hide the fact that we scanned duplicates files